# TRAINING DATA CNT


### JSON파일 하나만 출력

In [11]:


import json
with open("C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease/V006_77_1_07_04_01_13_1_3321w_20201229_119.jpg.json", "r", encoding="utf8") as f:
    contents = f.read()
    json_data = json.loads(contents)

disease = json_data["annotations"]["disease"]
risk = json_data["annotations"]["risk"]
print(disease)
print(risk)

7
1


## 질병데이터 확인

In [13]:
import os 
import json
from glob import glob

file_list = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

for file in file_list:
    with open(file_list[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [17]:
file_list = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease/*.json')
file_list

['C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_1_3321w_20201229_119.jpg.json',
 'C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_2_1886w_20201209_30.jpg.json',
 'C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_2_3108w_20201118_2.jpg.json',
 'C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_2_3108w_20201210_33.jpg.json',
 'C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_2_3248b_20201202_21.jpg.json',
 'C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_disease\\V006_77_1_07_04_01_13_2_4660w_20201202_23.jpg.json',
 'C:/Users/2018A00591/

In [22]:
risk_B1_cnt+risk_B2_cnt+risk_B3_cnt

risk_A1_cnt+risk_A2_cnt+risk_B3_cnt

1811

In [29]:
print("질병 데이터 폴더 갯수 확인 ")
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

질병 데이터 폴더 갯수 확인 
잿빛(7): 955 초기: 564 중기: 160 말기: 231
흰가루(8): 856 초기: 408 중기: 334 말기: 114
질병 데이터 토탈 :  1811


# 데이터 증강 갯수 확인

In [44]:
#루트 정상 

import json
with open("C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_augmentation/V006_77_1_07_04_01_13_1_3321w_20201229_119_a0000.jpg.json", "r", encoding="utf8") as f:
    contents = f.read()
    json_data = json.loads(contents)

disease = json_data["annotations"]["disease"]
risk = json_data["annotations"]["risk"]
print(disease)
print(risk)

7
1


In [47]:
import os 
import json
from glob import glob

file_list_aug = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_augmentation/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

ob_cnt = 0 #정상
for file in file_list_aug:
    with open(file_list_aug[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob_cnt +=1
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [45]:
file_list_aug = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_augmentation/*.json')
len(file_list_aug)

18110

In [48]:
print("증강 데이터 폴더 갯수 확인 ")
print("정상: ", ob_cnt)
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

증강 데이터 폴더 갯수 확인 
정상:  0
잿빛(7): 9550 초기: 5640 중기: 1600 말기: 2310
흰가루(8): 8560 초기: 4080 중기: 3340 말기: 1140
질병 데이터 토탈 :  18110


# 정상데이터 확인하기

In [77]:
import os 
import json
from glob import glob

file_list_normal = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_nomal/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

ob_cnt = 0 #정상
for file in file_list_normal:
    with open(file_list_normal[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob_cnt +=1
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [78]:
print("정상 데이터 폴더 갯수 확인 ")
print("정상: ", ob_cnt)
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

정상 데이터 폴더 갯수 확인 
정상:  19679
잿빛(7): 0 초기: 0 중기: 0 말기: 0
흰가루(8): 0 초기: 0 중기: 0 말기: 0
질병 데이터 토탈 :  0


In [54]:
file_list_normal = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Training/labelingdata/04Strawberry/04Strawberry_nomal/*.json')
len(file_list_normal)

19679

* * *

# VALIDATION DATA CNT

## 질병데이터 확인

In [59]:
import os 
import json
from glob import glob

file_list = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_disease/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

for file in file_list:
    with open(file_list[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [58]:
file_list = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_disease/*.json')
len(file_list)

226

In [60]:
print("질병 데이터 폴더 갯수 확인 ")
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

질병 데이터 폴더 갯수 확인 
잿빛(7): 117 초기: 52 중기: 29 말기: 36
흰가루(8): 109 초기: 55 중기: 44 말기: 10
질병 데이터 토탈 :  226


## 증강데이터 확인

In [63]:
import os 
import json
from glob import glob

file_list_aug = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_augmentation/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

ob_cnt = 0 #정상
for file in file_list_aug:
    with open(file_list_aug[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob_cnt +=1
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"  
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [65]:
file_list = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_augmentation/*.json')
len(file_list)

2260

In [64]:
print("증강 데이터 폴더 갯수 확인 ")
print("정상: ", ob_cnt)
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

증강 데이터 폴더 갯수 확인 
정상:  0
잿빛(7): 1170 초기: 520 중기: 290 말기: 360
흰가루(8): 1090 초기: 550 중기: 440 말기: 100
질병 데이터 토탈 :  2260


## 정상데이터 확인

In [79]:
import os 
import json
from glob import glob

file_list_normal = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_nomal/*.json')

i = 0
ob = 0
st = ""
#1,2,3 초중말 #A잿빛 #B 흰가루
disease_cnt_A = 0
disease_cnt_B = 0

risk_A1_cnt = 0
risk_A2_cnt = 0
risk_A3_cnt = 0

risk_B1_cnt = 0
risk_B2_cnt = 0
risk_B3_cnt = 0

ob_cnt = 0 #정상
for file in file_list_normal:
    with open(file_list_normal[i],'r') as f:
        json_data = json.load(f)
 
    disease = json_data['annotations']['disease']
    risk = json_data['annotations']['risk']
    if disease == 0:
        ob_cnt +=1
        ob = 0
    elif disease == 7:
        disease_cnt_A +=1
        if risk == 1:
            risk_A1_cnt +=1
            ob = "잿빛: 딸초"
        elif risk == 2:
            risk_A2_cnt +=1
            ob = "잿빛: 딸중"
        elif risk == 3:
            risk_A3_cnt +=1
            ob = "잿빛: 딸말"
    elif disease == 8:
        disease_cnt_B +=1
        if risk == 1:
            risk_B1_cnt +=1
            ob = "흰가루: 딸초"
        elif risk == 2:
            risk_B2_cnt +=1
            ob = "흰가루: 딸중"
        elif risk == 3:
            risk_B3_cnt+=1
            ob = "흰가루: 딸말"
    f.close()
    
    i = i + 1
    
disease_total_cnt = disease_cnt_A + disease_cnt_B

In [80]:
print("정상 데이터 폴더 갯수 확인 ")
print("정상: ", ob_cnt)
print("잿빛(7): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_A,risk_A1_cnt,risk_A2_cnt,risk_A3_cnt))
print("흰가루(8): {} 초기: {} 중기: {} 말기: {}".format(disease_cnt_B,risk_B1_cnt,risk_B2_cnt,risk_B3_cnt))
print("질병 데이터 토탈 : ",disease_total_cnt)

정상 데이터 폴더 갯수 확인 
정상:  178
잿빛(7): 0 초기: 0 중기: 0 말기: 0
흰가루(8): 0 초기: 0 중기: 0 말기: 0
질병 데이터 토탈 :  0


In [73]:
file_list_normal = glob('C:/Users/2018A00591/Desktop/071.시설 작물 질병 진단/딸기 데이터/Validation/labelingdata/04Strawberry/04Strawberry_nomal/*.json')
len(file_list_normal)

178

 * * *

# Training Data  
## #총합정리#  
정상 : 19679장  
잿빛 : (10505장) // 초기 : 6204장 // 중기 : 1760장 // 말기 : 2541장  

흰가루 : (9416장) // 초기 : 4488장 // 중기 : 3674장 // 말기 : 1254 장  

## ##상세 분류##  
### 질병 데이터 폴더 갯수 확인    
잿빛(7): 955 초기: 564 중기: 160 말기: 231  
흰가루(8): 856 초기: 408 중기: 334 말기: 114  
질병 데이터 토탈 :  1811  

### 증강 데이터 폴더 갯수 확인   
정상:  0 질병 : 19921  
잿빛(7): 9550 초기: 5640 중기: 1600 말기: 2310  
흰가루(8): 8560 초기: 4080 중기: 3340 말기: 1140  
질병 데이터 토탈 :  18110  
  
### 정상 데이터 폴더 갯수 확인   
정상:  19679  
잿빛(7): 0 초기: 0 중기: 0 말기: 0  
흰가루(8): 0 초기: 0 중기: 0 말기: 0  
질병 데이터 토탈 :  0  

# Validation Data  
## #총합정리#  
정상 : 178장 질병 : 2486장  
잿빛 : (1287장) 초기 : 572장 // 중기 : 319장 // 말기 : 396장  
  
흰가루 : (1199장) 초기 : 605장 // 중기 : 484장 // 말기 : 110 장  
  
### 질병 데이터 폴더 갯수 확인   
잿빛(7): 117 초기: 52 중기: 29 말기: 36  
흰가루(8): 109 초기: 55 중기: 44 말기: 10  
질병 데이터 토탈 :  226  
  
### 증강 데이터 폴더 갯수 확인   
정상:  0  
잿빛(7): 1170 초기: 520 중기: 290 말기: 360  
흰가루(8): 1090 초기: 550 중기: 440 말기: 100  
질병 데이터 토탈 :  2260  
  
### 정상 데이터 폴더 갯수 확인   
정상:  178  
잿빛(7): 0 초기: 0 중기: 0 말기: 0  
흰가루(8): 0 초기: 0 중기: 0 말기: 0  
질병 데이터 토탈 :  0  